### PR TITLE
Add displayProductPriceBlock hook

### PR DESCRIPTION
**Pull request by idnovate**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Added new hook to display information in the checkout summary. This hook is used everytime that product price is displayed but in this template
| Type?         | improvement
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| How to test?  | See #17295. Create a module and hook it to `displayProductPriceBlock`. Then display any information. Example module provided in the comments below

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/17296)

### CODE REVIEW

**Review by matthieu-rolland — COMMENTED** [review]
LGTM for me, codewise, but no issue has been created for this, I would like the opinion of other core developers before validating this PR ?

@PrestaShop/prestashop-core-developers what do you think?

File: themes/classic/templates/checkout/_partials/cart-summary-products.tpl [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ *}
+
+ <div class="cart-summary-products">
+    <p>{$cart.summary_string}</p>
```
Comment (PierreRambaud) [inline themes/classic/templates/checkout/_partials/cart-summary-products.tpl]: Could you use two spaces for indentation please? :)

File: themes/classic/templates/checkout/_partials/cart-summary-products.tpl [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    <p>
+        <a href="#" data-toggle="collapse" data-target="#cart-summary-product-list">
+            {l s='show details' d='Shop.Theme.Actions'}
+            <i class="material-icons">expand_more</i>
+        </a>
+    </p>
+
+    {block name='cart_summary_product_list'}
+    <div class="collapse" id="cart-summary-product-list">
```
Comment (PierreRambaud) [inline themes/classic/templates/checkout/_partials/cart-summary-products.tpl]: wrong indent here

File: themes/classic/templates/checkout/_partials/cart-summary-products.tpl [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        <ul class="media-list">
+            {foreach from=$cart.products item=product}
+            <li class="media">{include file='checkout/_partials/cart-summary-product-line.tpl' product=$product}</li>
```
Comment (PierreRambaud) [inline themes/classic/templates/checkout/_partials/cart-summary-products.tpl]: wrong indent here

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
Little indentation problem, and everything's ok for me

**Comment by idnovate** [comment]
> Little indentation problem, and everything's ok for me

Done

**Review by matthieu-rolland — DISMISSED** [review]
lgtm, thank you @idnovate 👍

File: themes/classic/templates/checkout/_partials/cart-summary-products.tpl [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+  <p>
+    <a href="#" data-toggle="collapse" data-target="#cart-summary-product-list">
+      {l s='show details' d='Shop.Theme.Actions'}
+      <i class="material-icons">expand_more</i>
+    </a>
+  </p>
+
+  {block name='cart_summary_product_list'}
+  <div class="collapse" id="cart-summary-product-list">
```
Comment (PierreRambaud) [inline themes/classic/templates/checkout/_partials/cart-summary-products.tpl]: need one more indentation level :)

Comment (idnovate) [reply themes/classic/templates/checkout/_partials/cart-summary-products.tpl]: You are right!

**Review by PierreRambaud — DISMISSED** [review]

**Review by matthieu-rolland — DISMISSED** [review]

**Review by matks — DISMISSED** [review]

**Comment by matks** [comment]
@idnovate Can you provide a test module for QA team ? If you have not enough time, I can do it 😉

**Comment by idnovate** [comment]
Please install attached module to test it.

image

[dummymodule.zip](https://github.com/PrestaShop/PrestaShop/files/4160243/dummymodule.zip)

**Comment by matks** [comment]
Great ! Then I pass the PR to QA team 😄

**Comment by SD1982** [comment]
Hi @idnovate could you please resolve the conflicting files ?

**Comment by idnovate** [comment]
> Hi @idnovate could you please resolve the conflicting files ?

Done :thumbsup:

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @idnovate,

Thanks, it is OK.
With PR:
image

Without PR:
image

Thanks!

**Comment by Progi1984** [comment]
Thanks @idnovate

**Pull request merged**